app/models/concerns/permalinkable.rb

```diff
@@ -9,14 +9,13 @@ module Permalinkable
   end
 
   module ClassMethods
+    # find by permalink or id
+    def find_by_param(param)
+      where("permalink = ? OR id = ?", param, param).first
```

Comment:
If you have an `OR` statement like this, does it still take advantage of indices?